### PR TITLE
Issue #12671: Resolve pitest suppression for FinalLocalVariableCheck

### PR DESCRIPTION
**Pull request by coderdeadpool**
Related to #12671 #12341

Diff Regression config: https://gist.githubusercontent.com/coderdeadpool/96fa62c511add1eb980efe8de851256d/raw/57420e4b601c7ed073340dd039e5c4af7a0d1ccd/gistfile1.txt

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java [hunk 1]
```diff
@@ -724,7 +724,7 @@ private static boolean isVariableInForInit(DetailAST variableDef) {
      */
     private static boolean isInAbstractOrNativeMethod(DetailAST ast) {
         boolean abstractOrNative = false;
-        DetailAST parent = ast.getParent();
+        DetailAST parent = ast;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java:727]: `DetailAST currentAst = ast;`

**Review by romani — CHANGES_REQUESTED** [review]
items:

**Comment by romani** [comment]
please share regression report

**Comment by coderdeadpool** [comment]
Github, generate repot

**Comment by coderdeadpool** [comment]
Github, generate report

**Comment by romani** [comment]
use all tokens at `property name="tokens" value="VARIABLE_DEF"`

**Comment by coderdeadpool** [comment]
Done

**Comment by nrmancuso** [comment]
@coderdeadpool you must have only a single ` ` between `Diff Regression config:` and link, newlines are not supported

**Comment by nrmancuso** [comment]
Github, generate report

**Comment by nrmancuso** [comment]
@coderdeadpool report generation failure here is from out of memory error in CI, this is usually an indicator of a lot of diffs/violations. Please generate report locally with all projects and share via github.io

**Comment by romani** [comment]
Or split config in two configs and execute report in two trigger comments, one after another.

**Comment by romani** [comment]
Please do not share images of reports, do uploading  to GitHub, read on how to do this at https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/README_MANUAL_EXECUTION.md#executing-diffgroovy

**Comment by coderdeadpool** [comment]
Github, generate website

**Comment by nrmancuso** [comment]
Github, generate report

**Comment by romani** [comment]
@coderdeadpool, please remove conflict to let me merge

**Comment by coderdeadpool** [comment]
Done.

**Review by romani — APPROVED** [review]
Ok to merge if CI pass

**Comment by coderdeadpool** [comment]
@romani Do we have to generate report again ?

**Comment by romani** [comment]
No, it was conflict in suppression file

**Pull request merged**